### DISCUSSION TITLE
Related to integration of concore-editor and concore

### DISCUSSION THREAD

**Discussion by parteekcoder** [ControlCore-Project/concore-editor#137] [Q&A] [answered]
@pradeeban 

We are also keeping concore flask server locally ?

**Comment by pradeeban** [comment, marked as answer]
That is correct.

concore (with FRI) runs locally. 
concore editor - for most of the time, we run it just from the browser, invoking the github.io version. We don't usually run concore editor from localhost.

**Reply by parteekcoder** [reply]
it means user will start the concore server locally , right?

Which I think there should be some easy way (like some GUI to start flask server) to start it as not all people know tech, what do you think if i am saying right?

**Reply by pradeeban** [reply]
Yes, that is a valid point. We should think about that. As of now, user action is needed to manually start the FRI, which can be annoying for a user who is not familiar with these technologies. (We have users who are neuroscientists, and we don't want to annoy them with the requirement to "Start FRI (i.e., the concore server)"

**Reply by parteekcoder** [reply]
yeah the solution can be to install concore as a software and With (at the beginning)minimal GUI of starting and stopping server

**Reply by pradeeban** [reply]
Sounds good to me. An installer that can install concore and concore-editor and then start both "concore server" (i.e., the FRI) and concore editor can be an option.

Another option is to just install concore FRI and start as a service, with loading the concore-editor as we do now.

A third option is, to start the FRI from a browser-based invocation from concore-editor itself (no installation required).

A local installation of concore editor can have its cons too - as that may trigger more bugs. As of now, as we use the github hosted version of concore editor, we can be confident that the critical bugs are minimal.

**Reply by parteekcoder** [reply]
I think good solution be is 
- concore-editor run as github hosted version
- concore can be installed as service
 Beacause of reason that Concore-editor
- is in its development stages
- There is more patches we introduce daily so difficult for users to install new version , In future it can be installed when it is in stable versions
- We are increasing features of concore-editor as enhancement

**Reply by pradeeban** [reply]
Sounds reasonable. At some point, when we are really confident of the status of concore editor, we can bundle it as a software package together with concore (and FRI). Perhaps, we can give the users the option. If they like a local version, they can use this "bundle." Otherwise, they can follow the third (or second) option I mentioned.

**Reply by Rahuljagwani** [reply]
In my opinion 3rd option is best (if possible effectively ) as user don't have to worry about any unnecessary installation. For this, the simplest and most effective solution according to me is to host the FRI flask server permanently on any hosting or cloud service like heroku etc. (Unfortunately heroku is paid now)

@pradeeban  @parteekcoder 
I request you to comment on this approach and add any disadvantages or problems this approach can face.

**Reply by pradeeban** [reply]
Yes, because assuming all the Node versions will work seamlessly is tricky. (Even we have not figured out the line ending issue in the local execution for the most part). Keeping the concore editor run from github server is smooth (and an approach followed by many web-based graph editors too).
